drivers/interrupt_controller/intc_scr_ipic.c

```diff
@@ -0,0 +1,209 @@
+/**
+ * Copyright (c) 2023 Syntacore
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * @file intc_scr_ipic.c
+ * @author psk-sc
+ * @brief Syntacore Integrated Programmable Interrupt Controller (IPIC) Interface
+ *        for RISC-V processors
+*/
+
+#define DT_DRV_COMPAT riscv_scr_ipic
```

Comment:
Why the `riscv` compatible prefix? Shouldn't this be `syntacore` or something like that?